packages/runtime/src/helpers/utilities/functions.ts

```diff
@@ -0,0 +1,43 @@
+import { existsSync, readJSON, writeFile } from 'fs-extra'
+import { join } from 'pathe'
+
+import { NEXT_PLUGIN, NEXT_PLUGIN_NAME } from '../../constants'
+import { resolveModuleRoot } from '../config'
+
+const checkForPackage = async (packageDir: string, nodeModule: boolean) => {
+  const packagePlugin = existsSync(packageDir) ? await readJSON(packageDir) : null
+  let nextPlugin
+  if (!nodeModule && packagePlugin) {
+    nextPlugin = packagePlugin.dependencies[NEXT_PLUGIN] ? packagePlugin.dependencies[NEXT_PLUGIN] : null
+  } else if (nodeModule && packagePlugin) {
+    nextPlugin = packagePlugin.version ? packagePlugin.version : null
+  }
+
+  return nextPlugin
+}
+
+/**
+ * Creates a function configuration file for the given function
+ *
+ * @param functionName The name of the function, e.g. `___netlify-handler`
+ * @param functionTitle The name of the function that will be displayed in logs, e.g. `Next.js SSR handler`
+ * @param functionsDir The directory where the function is located, e.g. `.netlify/functions`
+ */
+export const writeFunctionConfiguration = async (functionName: string, functionTitle: string, functionsDir: string) => {
+  const pluginPackagePath = '.netlify/plugins/package.json'
+  const ProjDir = resolveModuleRoot(NEXT_PLUGIN)
+  const nodeModulesPath = `${ProjDir}/package.json`
+
+  const nextPluginVersion =
+    (await checkForPackage(nodeModulesPath, true)) || (await checkForPackage(pluginPackagePath, false))
+
+  const metadata = {
+    config: {
+      name: functionTitle,
+      generator: nextPluginVersion ? `${NEXT_PLUGIN_NAME}@${nextPluginVersion}` : 'Next Runtime Version Not Found',
+    },
+    version: 1,
+  }
+
+  await writeFile(join(functionsDir, functionName, `${functionName}.json`), JSON.stringify(metadata))
```

Comment:
If we have a function `functionsDir/function.ts` then this is writing to `functionsDir/functionName/functionName.json`. Is this expected?
I think it's adding an extra directory that we don't need.